pil-analyzer/src/statement_processor.rs

```diff
@@ -224,15 +225,17 @@ where
         &mut self,
         PolynomialName { name, array_size }: PolynomialName,
         base_type: Type,
-    ) -> (String, Option<TypeScheme>) {
+    ) -> Result<(String, Option<TypeScheme>), Error> {
         let ty = Some(match array_size {
             None => base_type.into(),
             Some(len) => {
+                let source_ref = len.source_reference().clone();
                 let len = self
                     .expression_processor(&Default::default())
                     .process_expression(len)
-                    // TODO propagate this error up
-                    .expect("Failed to process length expression");
+                    .map_err(|err| {
+                        source_ref.with_error(format!("Failed to process length expression: {err}"))
```

Comment:
If `err` already has a more specific source reference, that one is lost in here, isn't it? Could you check how the individual errors look at the end?